core/src/main/scala/org/ensime/indexer/graph/GraphService.scala

```diff
@@ -40,30 +97,66 @@ object FileCheck extends ((String, Timestamp) => FileCheck) {
     FileCheck(name, ts)
   }
 }
-final case class FqnSymbol(
-    id: Option[Int],
-    file: String, // the underlying file
-    path: String, // the VFS handle (e.g. classes in jars)
-    fqn: String,
-    internal: Option[String], // for fields
-    source: Option[String], // VFS
-    line: Option[Int],
-    offset: Option[Int] = None // future features:
-//    type: ??? --- better than descriptor/internal
-) {
-  // this is just as a helper until we can use more sensible
-  // domain objects with slick
-  def sourceFileObject(implicit vfs: EnsimeVFS) = source.map(vfs.vfile)
-
-  // legacy: note that we can't distinguish class/trait
-  def declAs: DeclaredAs =
-    if (fqn.contains("(")) DeclaredAs.Method
-    else if (internal.isDefined) DeclaredAs.Field
-    else DeclaredAs.Class
-}
 
 // core/it:test-only *Search* -- -z prestine
 class GraphService(dir: File) extends SLF4JLogging {
+  import org.ensime.indexer.graph.GraphService._
+
+  implicit object TimeStampSPrimitive extends SPrimitive[Timestamp] {
```

Comment:
these `SPrimitive` impls can probably be moved into the stringymap package / files.